### ISSUE TITLE
bug: E2E beforeAll() error: The page was already closed

### ISSUE THREAD

**Issue by kraftwer1** [ionic-team/stencil#3720]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

2.18.1

### Current Behavior

The Jest function `beforeAll()` cannot be used to set up a `page` (and possibly even other suite-wide objects) . It seems to end after the first `it()` and won't work for the second `it()`, instead it throws the error `"The page was already closed"`.

Example:

```
describe("suite", () => {
  let page

  beforeAll(async () => {
    page = await newE2EPage()
  })

  it("1", () => {
    const someEl = await page.find('my-component >>> div') // works so far
  })

  it("2", () => {
    const someEl = await page.find('my-component >>> div') // error "The page was already closed" 
  })
})
```

It works fine when using `beforeEach()` instead of `beforeAll()`, however `beforeEach()` slows down tests a lot because `newE2EPage()` takes a lot of time and isn't required in many cases.

### Expected Behavior

`beforeAll()` works just like `beforeEach()`. According to the Jest docs, `beforeAll()` and its counterpart `afterAll()` are built for exactly that: https://jestjs.io/docs/setup-teardown

### Steps to Reproduce

Clone the `stencil-component-starter` repo and alter `my-component.e2e.ts` as described above.

### Code Reproduction URL

https://github.com/ionic-team/stencil-component-starter

**Comment by rwaskiewicz** [comment]
Hey @kraftwer1 👋 

Thanks for the bug report! I can confirm that this behavior does occur in Stencil and agree it's not what I would expect with using `beforeAll`. This appears to have been done in https://github.com/ionic-team/stencil/commit/66ad23d5bdb7b3b6ca36fb3c390eb05c53411c91 in order to reduce memory usage. I don't have much more context as to why that behavior is/was configured ATM. I'm going to label this for the team to take a closer look.

Thanks!

**Comment by tanner-reits** [comment]
A fix for this was included in today's [v4.19.0 release](https://github.com/ionic-team/stencil/releases/tag/v4.19.0)!
